game/hud/src/widgets/Compass/index.tsx

```diff
@@ -64,11 +79,20 @@ class Compass extends React.Component<CompassProps, CompassState> {
   }
 
   render() {
+    if (!this.state.visible) return null;
```

Comment:
visibility should be handled by the HUD layout module, not by an individual widget.